### PR TITLE
Error Handling for Show Discounts [P0]

### PR DESCRIPTION
**Pull request by khushi2305**
This PR performs the following actions  - 
1) Handled erroneous  cases . Eg when the access token is not set or  devices/structure list returned is empty/null etc 
In such cases no change will occur on the screen . Later snackbars  could be displayed for them showing the error to the user . 
1) All the discounts are disabled initially . They get enabled only after the user chooses the home and valid offers are displayed . The disabling is executed  everytime the user navigates to a different page to ensure that the discounts can be selected only when they eligible ones are displayed  .
2) Now clicking the same discount again would reverse its state . That is , if it is selected , it will get deselected and vice a versa . 
3) A button is added to the Show Discounts page for directly getting the list of structures ( untill the access token is valid ) so that the user does not have to go through the entire process of resource picker repeatedly . 
Tested manually .
image

### CODE REVIEW

**Comment by khushi2305** [comment]
@googlebot I consent.

**Comment by dreamplay** [comment]
please add details about TESTING in PR description including unit tests/ integration tests / manually tested as applicable

File: home_insurance_app/lib/data/helper_functions.dart [hunk 1]
```diff
@@ -10,18 +10,33 @@ Future<List> getAllowedOffers(BuildContext context) async {
...
+  }
+  // In case authorisation is not successful or structure is empty , empty list is returned , else list with desired offers is returned
+  return (allowedOffers);
+}
 
-// Helper function to show dialogue box for displaying structure list
+Future<List> selectStructure(BuildContext context) async {
+  List<Offer> allowedOffers = [];
+  Optional<List> response;
+  // Try catch is applied to catch exceptions thrown by sdk  when access token is not set
```
Comment (dreamplay) [inline home_insurance_app/lib/data/helper_functions.dart:22]: remove this comment

Comment (khushi2305) [reply home_insurance_app/lib/data/helper_functions.dart:22]: Done

**Comment by khushi2305** [comment]
@googlebot I consent.

**Comment by khushi2305** [comment]
@googlebot I consent.

**Comment by khushi2305** [comment]
@googlebot I consent.

**Comment by oshhh** [comment]
@googlebot I consent.

**Comment by anjalisoni3655** [comment]
@googlebot I consent.

**Comment by anjalisoni3655** [comment]
@googlebot I consent.

**Comment by khushi2305** [comment]
@googlebot I consent.

**Comment by oshhh** [comment]
@googlebot I consent.

**Comment by oshhh** [comment]
@googlebot I consent.

**Comment by khushi2305** [comment]
@googlebot I consent.

**Pull request merged**